b_cfn_elasticsearch_cloner/cloner_source/index.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import os
+from typing import Any, Dict
+
+import boto3
+from elasticsearch import Elasticsearch, RequestsHttpConnection
+from elasticsearch.helpers import bulk
+from requests_aws4auth import AWS4Auth
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+try:
+    from dynamodb_decode import DynamodbDecoder
+except ImportError as ex:
+    logger.exception("Failed import.")
+    from b_cfn_elasticsearch_cloner.cloner_source.dynamodb_decode import DynamodbDecoder
+
+
+es_index = os.environ["ES_INDEX_NAME"]
+es_endpoint = os.environ["ES_DOMAIN_ENDPOINT"]
+primary_key_field = os.environ["PRIMARY_KEY_FIELD"]
+region = os.environ["AWS_REGION"]
+
+
+credentials = boto3.Session().get_credentials()
+awsauth = AWS4Auth(
+    credentials.access_key,
+    credentials.secret_key,
+    region,
+    "es",
+    session_token=credentials.token,
+)
+
+es = Elasticsearch(
+    hosts=[{"host": es_endpoint, "port": 443}],
+    http_auth=awsauth,
+    use_ssl=True,
+    verify_certs=True,
+    connection_class=RequestsHttpConnection,
+)
+
+def handler(event: Dict[Any, Any], context: Any) -> Dict[str, Any]:
+    logger.info("Starting processing DynamoDB events.")
+
+    succeeded, failed = bulk(
+        es,
+        dynamodb_to_es_generator(event),
+        stats_only=True,
+        raise_on_error=False,
+        raise_on_exception=False,
+    )
+
+    logger.info(f"Finished processing DynamoDB events. Succeeded: {succeeded}, failed: {failed}")
+
+def dynamodb_to_es_generator(event: Dict[Any, Any]):
```

Comment:
This seems to be a pretty significant function too. Please add a complete docstring that tells what this function does. Additional References and Examples would be great.